files/class-vip-filesystem-stream-wrapper.php

```diff
@@ -702,8 +702,8 @@ public function stream_metadata( $path, $option, $value ) {
 				if ( false === file_exists( $path ) ) {
 					$file = fopen( $path, 'w' );    // phpcs:ignore WordPress.WP.AlternativeFunctions.file_system_read_fopen
 					if ( is_resource( $file ) ) {
-						fclose( $file );
-						return true;
+						$result = fflush( $file );
+						return fclose( $file ) && $result;
```

Comment:
Now the most interesting part: PHP [does not report errors from `fclose()`](https://heap.space/xref/PHP-8.0/ext/standard/file.c?r=7e947908#910). Even if `stream_close()` fails, `fclose()` will return `true`.

This is why we have to call `flush()` to make sure we catch the error if the file upload fails, and then return the proper result.